tests/TestCase/Datasource/SimplePaginatorTest.php

```diff
@@ -55,39 +55,39 @@ public function testPaginateCustomFind(): void
         $this->assertEquals(['First Post', 'Second Post', 'Third Post', 'Fourth Post'], $titleExtractor($result));
 
         $pagingParams = $this->Paginator->getPagingParams();
-        $this->assertSame(4, $pagingParams['PaginatorPosts']['current']);
-        $this->assertNull($pagingParams['PaginatorPosts']['count']);
+        $this->assertSame(1, $pagingParams['PaginatorPosts']['currentPage']);
+        $this->assertNull($pagingParams['PaginatorPosts']['totalCount']);
 
         $settings = ['finder' => 'published'];
         $result = $this->Paginator->paginate($table, [], $settings);
         $this->assertCount(3, $result, '3 rows should come back');
         $this->assertEquals(['First Post', 'Second Post', 'Third Post'], $titleExtractor($result));
 
         $pagingParams = $this->Paginator->getPagingParams();
-        $this->assertSame(3, $pagingParams['PaginatorPosts']['current']);
-        $this->assertNull($pagingParams['PaginatorPosts']['count']);
+        $this->assertSame(1, $pagingParams['PaginatorPosts']['currentPage']);
+        $this->assertNull($pagingParams['PaginatorPosts']['totalCount']);
```

Comment:
I like the new names they are clearer. If we're adding new names could we also backport them to 4.x so that folks who rely on these keys have a chance to upgrade early?

Reply:
IMO that would cause unnecessary grief to 4.x users who are directly dealing with the paging params. Array keys changes unlike API change are not easy to communicate. The `count` for e.g. now holds the resultset count instead of total count and cause bugs in existing code. 

There are community plugins too which use these keys:

https://github.com/FriendsOfCake/crud/blob/ace4262d8c1c6d4b26937cb19f8d3b1b954aeb4b/src/Listener/ApiPaginationListener.php#L55-L60

As a user I would rather upgrade these keys when I upgrade to next major instead of having to do that now.

Reply:
Is there a way we could backport the new implementation and have users opt-in to the new implementation using config options? I'm trying to figure out how we make the 5.x upgrade process like the 4.x one, where once you had fixed deprecation warnings you didn't have much else to do.

Reply:
Not having to deal with the pain of trying to have 2 implementations together is why I did this for 5.x. For majority of the users these would be transparent. The only change they would have to do when upgrading is explicitly set the the paginated resultset to the paginator helper.